src/olcs/Service.js

```diff
@@ -41,6 +81,81 @@ const Service = class {
   getMaxTilt() {
     return 7 * Math.PI / 16;
   }
+
+  /**
+   * @private
+   */
+  cameraFromPermalink_() {
```

Comment:
I discussed with @sbrunner about how to implement and where to put the code for this new feature.
The whole picture is:

A  `mapState` service should be created with the current code for 2d.
The code for handling 3d should be added in the same file.
The `mapState` deals with reading and writting the 2d view and 3d camera (not other parameters like the layers).
The `mapState` should not know about local storage or permalink.

Reply:
I pushed a change using `ngeoStateManager` instead of `ngeoLocation`. I think this cover the need for this project.
Shall we start the big refactoring under Luxembourg fundings ?
I think it's a big work and i'm not sure about the whole approach.

> The mapState should not know about local storage or permalink.

What do you mean ?